### DISCUSSION TITLE
[Info] We built a mobile 📱 AND WINDOWS 🪟  interactive version using flutter and it runs super well on a simple Oneplus 7 with 8GB ram.

### DISCUSSION THREAD

**Discussion by ThibautLEAUX** [ggml-org/llama.cpp#568] [Show and tell]
# Have fun

Hi, it is not a real issue, but i want to share here that we made a running app with interactive mode on mobiles using your repo.
You can find our repo here : [https://github.com/Bip-Rep/sherpa](https://github.com/Bip-Rep/sherpa)

Unfortunately it doesnt have your latest commit because there is an error during runtime but we made a fork here [https://github.com/Bip-Rep/llama.cpp](https://github.com/Bip-Rep/llama.cpp) You need to be on the "for_mobile" branch to build the libraries. We are using an older working commit on this branch.

We translated the main functions of llama.cpp in dart.

## Working demo
[![IMAGE ALT TEXT HERE](https://img.youtube.com/vi/jdw7oABjTeQ/0.jpg)](https://www.youtube.com/watch?v=jdw7oABjTeQ)
Click on the image to view the video on YouTube.

Hope this helps.

**Comment by anzz1** [comment]
Looks really cool ! :+1: 
This definitely captures the essence of "inference on the edge", what this project is all about. :smile: 

This would be otherwise moved to a discussion but if there is a runtime error on the newer commits, well that definitely sounds like a issue. Could you elaborate on that?

**Reply by ThibautLEAUX** [reply]
Thanks !!

We have a hard crash without log.. we are trying to have a starcktrace but we dont have it for now, thats why i did not make a templated issue..
I hope someone finds a way

**Comment by gjmulder** [comment]
If there is an issue please log an issue with the [issue template](https://github.com/ggerganov/llama.cpp/blob/master/.github/ISSUE_TEMPLATE/custom.md?plain=1)

**Reply by ThibautLEAUX** [reply]
As soon as we can have a log well do it, for now the crash doesnt provide a stack trace..

**Comment by ThibautLEAUX** [comment]
btw, we just released a new version of our flutter code that includes Windows build !!
[https://github.com/Bip-Rep/sherpa/releases/tag/2.0](https://github.com/Bip-Rep/sherpa/releases/tag/2.0)
